llvm/docs/HowToCrossCompileLLVM.rst

```diff
@@ -6,210 +6,202 @@ Introduction
+    set(CMAKE_C_FLAGS_INIT "$CFLAGS")
+    set(CMAKE_CXX_FLAGS_INIT "$CFLAGS")
+    set(CMAKE_LINKER_TYPE LLD)
+    set(CMAKE_C_COMPILER clang)
+    set(CMAKE_CXX_COMPILER clang++)
+    set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+    set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+    set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+    set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+    EOF
+
+To customise details of the compilation target or choose a different
+architecture altogether, just change the ``SYSROOT``,
+``TARGET``, and ``CFLAGS`` variables before copy and pasting the rest (e.g.
+for RISC-V you might set ``SYSROOT=$HOME/sysroot-deb-riscv64-unstable``,
+``TARGET=riscv64-linux-gnu``, ``CFLAGS="-march=rva20u64"``).
+
+Then configure and build with ``cmake`` invocations such as the following:
 
    .. code-block:: bash
 
-     $ CC='clang' CXX='clang++' cmake -G Ninja <source-dir> -DCMAKE_BUILD_TYPE=<type> <options above>
-
-If you have ``clang``/``clang++`` on the path, it should just work, and special
-Ninja files will be created in the build directory. I strongly suggest
-you to run ``cmake`` on a separate build directory, *not* inside the
-source tree.
-
-To build, simply type:
+    cmake -G Ninja \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DLLVM_ENABLE_PROJECTS="lld;clang" \
+      -DCMAKE_TOOLCHAIN_FILE=$(pwd)/$TARGET-clang.cmake \
+      -DLLVM_HOST_TRIPLE=$TARGET \
+      -DCMAKE_INSTALL_PREFIX=$HOME/clang-$TARGET \
+      -S llvm \
+      -B build/$TARGET
+    cmake --build build/$TARGET
+
+The following options from the toolchain file and ``cmake`` invocation above
+are worth highlighting:
+
+ * ``CMAKE_SYSTEM_NAME``: Perhaps surprisingly, explicitly setting this
+   variable `causes CMake to set
+   CMAKE_CROSSCOMPIILING <https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html#variable:CMAKE_CROSSCOMPILING>`_.
+ * ``CMAKE_{C,CXX}_COMPILER_TARGET``: This will be used to set the
+   ``--target`` argument to ``clang``. The triple should match the triple used
+   within the sysroot (i.e. ``$SYSROOT/usr/lib/$TARGET`` should exist).
+ * ``CMAKE_FIND_ROOT_PATH_MODE_*``: These `control the search behaviour for
+   finding libraries, includes or binaries
+   <https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html#finding-external-libraries-programs-and-other-files>`_
+   and are set so as to avoid files for the host being used in the build.
+ * ``LLVM_HOST_TRIPLE``: Specifies the target triple the built LLVM will run
+   on, which also implicitly sets other defaults such as
+   ``LLVM_DEFAULT_TARGET_TRIPLE``.
+ * ``CMAKE_SYSROOT``: Gives the path to the sysroot containing libraries and
+   headers for the target, as generated earlier in this document.
+ * ``CMAKE_INSTALL_PREFIX``: Unnecessary if you don't intend to use the
+   ``install`` target. But setting it avoids installing binaries compiled for
+   the target across system directories on the host system.
```

Comment:
I would swap the order because everything else has been like "setting name: purpose" and this one is like "setting name: please stop reading this bullet point unless you are in this one specific situation".

Cleaner if we put the caveat last.

"Setting this avoids installing binaries compiled for the target system into system directories for the host system. It is not required unless you are going to use the ``install`` target."

No harm in folks learning what it is, it's not that complicated.

Reply:
I've applied this suggested edit.